File: python/src/blind150/arrays/containsDuplicate.py
from typing import List
from src.utils.logging_util import get_logger

logger = get_logger(__name__)
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        logger.info("Checking for duplicates in list: %s", nums)
        seen = set()
        for num in nums:
            if num in seen:
                logger.warning("Duplicate found: %s", num)
                return True
            seen.add(num)
        logger.info("No duplicates found.")
        return False

# Driver code
if __name__ == "__main__":
    solution = Solution()  # Create an instance of the Solution class
    nums = [1, 2, 3, 4, 5, 3]  # Example input list
    result = solution.hasDuplicate(nums)  # Call the method

File: java/src/main/java/dsa/blind150/arrays/ContainsDuplicate.java
package dsa.blind150.arrays;

import dsa.utils.LoggerUtil;
import org.slf4j.Logger;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    private static final Logger logger = LoggerUtil.getLogger();
    public boolean hasDuplicate(int[] nums) {
        logger.info("Checking for duplicates in array: {}", nums);
        Set<Integer> unique = new HashSet<>();
        for (int num : nums) {
            if (unique.contains(num)) {
                logger.warn("Duplicate found: {}", num);
                return true;
            }
            unique.add(num);
        }
        logger.info("No duplicates found.");
        return false;
    }

    public static void main(String[] args) {
        int[] hasDuplicates = {1, 2, 3, 1};
        int[] hasNoDuplicates = {1, 2, 3, 4};

        ContainsDuplicate solution = new ContainsDuplicate();

        System.out.println(solution.hasDuplicate(hasDuplicates)); // Output: true
        System.out.println(solution.hasDuplicate(hasNoDuplicates)); // Output: false
    }
}